.github/workflows/pr-gate.yml

```diff
@@ -7,9 +7,13 @@ name: pr-gate
 on:
   pull_request:
     branches: [ staging, main ]
```

Comment:
if this pr-gate.yml is the equivalent of the unit tests we had, could we please follow the same pattern and rename this as unit so it is more clear

Reply:
> @laserprec do you have the time saving between doing the old pipeline vs this one?

@miguelgfierro, need to fix some spark bugs on the self-hosted machine. Will get back to you later :)

Reply:
> if this pr-gate.yml is the equivalent of the unit tests we had, could we please follow the same pattern and rename this as unit so it is more clear

@miguelgfierro, if I understand correctly, you are suggesting that we can rename `pr-gate.yml` to `unit-test.yml`. Is that right? In that case, I think it would be best we stick with a more functional name describing utility of the pipeline like `gating a PR from merging` because in the future, depending on how we structure tests, we can possibly run smoke/e2e/integration tests in the PR-gate itself.